tools/src/main/scala/scala/scalanative/build/package.scala

```diff
@@ -0,0 +1,148 @@
+package scala.scalanative
+
+import java.nio.file.{Files, Path, Paths}
+import java.util.Arrays
+
+import scala.sys.process.Process
+
+import nir.Global
+
+import tools.{
+  Config,
+  GarbageCollector,
+  IO,
+  LinkerReporter,
+  LinkerResult,
+  Logger,
+  OptimizerDriver,
+  OptimizerReporter
+}
+import IO.RichPath
+
+package object build {
+
+  def build(nativeLib: Path,
+            paths: Seq[Path],
+            entry: String,
+            target: Path,
+            workdir: Path,
+            logger: Logger): Path = {
+    val config = Config.default(nativeLib, paths, entry, workdir, logger)
+    build(config, target, logger)
+  }
+
+  def build(config: Config, target: Path, logger: Logger) = {
+    val linkerResult = tools.link(config)
+    val optimized =
+      tools.optimize(config, linkerResult.defns, linkerResult.dyns)
+    val generated = {
+      tools.codegen(config, optimized)
+      IO.getAll(config.workdir, "glob:**.ll")
+    }
+    val objectFiles = llvm.compileLL(config, generated, logger)
+    val unpackedLib = unpackNativeLibrary(config.nativeLib, config.workdir)
+
+    val nativeLibConfig =
+      config.withCompileOptions("-O2" +: config.compileOptions)
```

Comment:
Passing an extra O2 option is duplicated here and in sbt plugin. There is a risk that it's changed in one place and not the other, lets refactor this.